File: Python Word examples/documentProperties/addCustomDocumentProperties.py
import spirecloudword
from spirecloudword import CustomDocumentProperty
from spirecloudword.configuration import Configuration as WordConfiguration

appId = "your id"
appKey = "your key"

configuration = WordConfiguration(appId, appKey)
api = spirecloudword.api.document_properties_api.DocumentPropertiesApi(configuration)

#Support file format: docx, doc, docm, dot, dotx, dotm, rtf
file_name = "Template.docx"
custom_properties_list = [CustomDocumentProperty("Keywords", "Set document properties."),
                          CustomDocumentProperty("Author", "eiceblue"), CustomDocumentProperty("Company", "e-iceblue"),
                          CustomDocumentProperty("LastSavedBy", "123456@iCloud.com"),
                          CustomDocumentProperty("SharedDoc", "true"), CustomDocumentProperty("Date", "2019/10/30")]
extra = {'folder': "input", 'dest_file_path': "/output/AddCustomDocumentProperties.docx"}

result = api.add_custom_document_properties(file_name, custom_properties_list, **extra)

File: JavaScript Word Examples/DocumentPropertiesApiDemo/addCustomDocumentProperties.js
(function (){
    var appId = "your id";
    var appKey = "your key"; 
    var baseUrl = "https://api.e-iceblue.cn";

    var Spirecloudword= require('../../src/index');
    var apiClient = new Spirecloudword.ApiClient(appId, appKey,baseUrl);
    var instance = new Spirecloudword.DocumentPropertiesApi(apiClient);

    var filename = "Template.docx";
    var properties = [new Spirecloudword.CustomDocumentProperty("Keywords", "Set document properties.", "string"),
                    new Spirecloudword.CustomDocumentProperty("Author", "eiceblue"),
                    new Spirecloudword.CustomDocumentProperty("Company", "冰蓝"),
                    new Spirecloudword.CustomDocumentProperty("LastSavedBy", "123456@iCloud.com"),
                    new Spirecloudword.CustomDocumentProperty("SharedDoc", "true", "boolean"),
                    new Spirecloudword.CustomDocumentProperty("Date", "2019/1/30", "datetime"),
                    new Spirecloudword.CustomDocumentProperty("Number", "8", "number"),];
    var destFilePath = "output/AddCustomDocumentProperties.docx";
    var opt = { 'folder': "input", 'properties': properties };
    instance.addCustomDocumentProperties(filename, properties, destFilePath, opt,
        function (error, data, response) {
            if (error) throw error;
        });
})();